server/controllers/configurationsController/configurationsController.test.js

```diff
@@ -0,0 +1,184 @@
+
+    describe('When successful', () => {
+      const request = createRequestWithUser(newConfiguration)
+      const response = createResponse()
+
+      it('sends a status of 200 with a data property when successful', async () => {
+        request.app.locals.appDb.insertOne.mockResolvedValueOnce({
+          insertedId: 'eagle',
+        })
+
+        await ConfigurationsController.create(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          data: 'eagle',
+        })
+      })
+    })
+
+    describe('When unsuccessful', () => {
+      it('sends a status of 400 with a message when database there is an error', async () => {
+        const request = createRequestWithUser(newConfiguration)
+        const response = createResponse()
+
+        request.app.locals.appDb.insertOne.mockImplementationOnce(() => {
+          insertedId: null
+        })
+
+        await ConfigurationsController.create(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(400)
+        expect(response.json).toHaveBeenCalledWith({
+          error: `Cannot read properties of undefined (reading 'insertedId')`,
+        })
+      })
+    })
+  })
+
+  describe(ConfigurationsController.edit, () => {
+    const configAttributes = { body: { name: 'new matrix', owner: 'owl' } }
+
+    describe('When successful', () => {
+      it('sends a status of 200 and with a data payload', async () => {
+        const request = createRequestWithUser(configAttributes)
+        const response = createResponse()
+
+        request.app.locals.appDb.findOneAndUpdate.mockResolvedValueOnce({
+          value: {},
+        })
+
+        await ConfigurationsController.edit(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          data: {},
+        })
+      })
+    })
+
+    describe('When not found', () => {
+      it('sends a status of 404 with a message when database fails to update configuration', async () => {
+        const request = createRequestWithUser(configAttributes)
+        const response = createResponse()
+
+        request.app.locals.appDb.findOneAndUpdate.mockResolvedValueOnce({
+          modified: 0,
+        })
+
+        await ConfigurationsController.edit(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(404)
+        expect(response.json).toHaveBeenCalledWith({
+          error: new Error('Unable to edit configuration'),
+        })
+      })
+    })
+
+    describe('When unsuccessful', () => {
+      it('sends a status of 400 with a message when database there is an error', async () => {
+        const request = createRequestWithUser(configAttributes)
+        const response = createResponse()
+
+        request.app.locals.appDb.findOneAndUpdate.mockImplementationOnce(() => {
+          new Error()
+        })
```

Comment:
```suggestion
        request.app.locals.appDb.findOneAndUpdate.mockRejectedValueOnce()
```

Reply:
[f11e811](https://github.com/AMP-SCZ/dpdash/pull/305/commits/f11e811a37bb7df8cc158a8d0e061615e8ec535d)

Reply:
Thanks for this. 

The mock still needs an error object with an error message. To go with the comment that was previously added by @taylorkearns. https://github.com/AMP-SCZ/dpdash/pull/305#discussion_r1223353767
The tests will look something like
`request.app.locals.appDb.findOneAndUpdate.mockRejectedValueOnce(new Error('some error'))`
to mock an error upon a rejected promise